R/simulate.r

```diff
@@ -387,25 +447,48 @@ simulate_summary <- function(index_cases, statistic = c("size", "length"),
         pars
       )
     )
-    if (any(next_gen %% 1 > 0)) {
-      stop("Offspring distribution must return integers")
+    # check that offspring distribution returns integers
+    stopifnot(
+      "Offspring distribution must return integers" =
+        !all(next_gen %% 1 > 0)
+    )
+    # Sample susceptible offspring to be infected from all possible offspring
+    # We first adjust for the case where susceptible can be Inf but prob is max
+    # 1.
+    binom_prob <- ifelse(
+      is.infinite(susc_pop),
+      1,
+      susc_pop / pop
+    )
```

Comment:
This is potentially being changed by #194.

More generally, are we generating a problem by ourselves by duplicating code? I think it would be a big improvement if we could ship some of these out to helper functions that are called both by `simulate_chains()` and `simulate_summary()` to reduce the amount of parallel code to an absolute minimum (ideally, 0) - where the only code in which they differ in the sampling and tracking of times.

Reply:
It is definitely problematic currently. I will link a new PR that addresses this comment and #44.

Reply:
For now, I've fixed this issue in 6753319769f2ece0b20786a40be1d616a16eb5b5. I'll merge this PR and fix the issue of shared code in a separate PR to avoid blowing up this PR.